### ISSUE TITLE
Incorrect batchone QueryRow method when using emit_result_struct_pointers

### ISSUE THREAD

**Issue by ericbock** [kyleconroy/sqlc#2203]
### Version

1.17.2

### What happened?

When using `:batchone` with `pgx/v4` and the `"emit_result_struct_pointers": true`, the `QueryRow` method does not use a pointer for the model when closed and calling the callback:

```go
func (b *GetAuthorBatchResults) QueryRow(f func(int, *Author, error)) {
	defer b.br.Close()
	for t := 0; t < b.tot; t++ {
		var i Author
		if b.closed {
			if f != nil {
				f(t, i, errors.New("batch already closed"))
			}
			continue
		}
		row := b.br.QueryRow()
		err := row.Scan(&i.ID, &i.Name, &i.Bio)
		if f != nil {
			f(t, &i, err)
		}
	}
}
```

Here, the callback expects an `*Author`, but it's getting `f(t, i, ...)` where `i` is an `Author`, not an `*Author`.

I think this should be

```go
		if b.closed {
			if f != nil {
				f(t, &i, errors.New("batch already closed"))
			}
			continue
		}
```

### Relevant log output

```shell
graph/store/batch.go:148:10: cannot use i (variable of type BatchGetSkusRow) as *BatchGetSkusRow value in argument to f
```

### Database schema

```sql
CREATE TABLE authors (
  id   BIGSERIAL PRIMARY KEY,
  name text      NOT NULL,
  bio  text
);
```

### SQL queries

```sql
-- name: GetAuthor :batchone
SELECT * FROM authors
WHERE id = $1 LIMIT 1;
```

### Configuration

```yaml
{
  "version": "2",
  "sql": [
    {
      "schema": "query.sql",
      "queries": "query.sql",
      "engine": "postgresql",
      "gen": {
        "go": {
          "out": "db",
          "sql_package": "pgx/v4",
	  "emit_result_struct_pointers": true
        }
      }
    }
  ]
}
```

### Playground URL

https://play.sqlc.dev/p/303a2261c50646777b637a96f62d00ede88494b8dda5cb88f3311eee3d9b8e99

### What operating system are you using?

macOS

### What database engines are you using?

PostgreSQL

### What type of code are you generating?

Go

**Comment by andrewmbenton** [comment]
Which version of sqlc are you using? When I run codegen on this example I get the following, so this may have been a bug that was previously fixed.

```go
		if b.closed {
			if f != nil {
				f(t, nil, ErrBatchAlreadyClosed)
			}
			continue
		}
```

**Comment by ericbock** [comment]
This is with v1.17.2

It looks like the batchCode template was updated recently (in https://github.com/kyleconroy/sqlc/commit/e505aec94bc471f5fcc7f16796ac9119a0adfb72), but besides defining an error type I don't see a change in the diff that would fix the problem.

**Comment by andrewmbenton** [comment]
Looking back before that most recent update to the template, there was a patch merged that addressed this issue: https://github.com/kyleconroy/sqlc/issues/1959

Based on the activity in that thread, I think the patch did not make it into the 1.17.2 release (it was merged one day after). But it should be in the next release.

**Comment by kyleconroy** [comment]
Fixed in v1.18.0
